mode/verilog/verilog.js

```diff
@@ -354,11 +360,189 @@ CodeMirror.defineMode("verilog", function(config, parserConfig) {
+            state.vxContext.svxCurCtlFlowChar  = "@";
+            stream.next();
+            stream.eatWhile(/[\w\$_]/);
+            break;
+          case /\b[mM]4+?/.test(curPunc):
+              // m4 pre proc
+              style = thisMode.svxkeywords[curPunc];
+            state.vxContext.svxCurCtlFlowChar  = "m4";
+            /* if (style=="def") */ stream.skipTo("(");
+            break;
+          case ch == "!" && stream.sol():
+            // v stmt in svx region
+            // state.vxContext.svxCurCtlFlowChar  = "S"; // bozo want to indent next tok
+            style = "comment";
+            stream.next();
+            break;
+          case thisMode.svxisOperatorChar.test(ch):
+            // operators
+            stream.eatWhile(thisMode.svxisOperatorChar);
+            style = "operator";
+            break;
+          case ch == "#":
+            // phy hier
+            state.vxContext.svxCurCtlFlowChar  = (state.vxContext.svxCurCtlFlowChar=="")
+              ? ch : state.vxContext.svxCurCtlFlowChar;
+            stream.next();
+            stream.eatWhile(/[+-]\d/);
+            style = "tag";
+            break;
+          case thisMode.svxkpScopePrefixs.propertyIsEnumerable(ch):
+            // special SVX operators
+            style = thisMode.svxkpScopePrefixs[ch];
+            state.vxContext.svxCurCtlFlowChar  = (state.vxContext.svxCurCtlFlowChar=="")
+              ? "S" : state.vxContext.svxCurCtlFlowChar;  // stmt
+            stream.next();
+            stream.match(/[a-zA-Z_0-9]+/);
+            break;
+          case thisMode.svxchScopePrefixs.propertyIsEnumerable(ch):
+            // special SVX operators
+            style = thisMode.svxchScopePrefixs[ch];
+            state.vxContext.svxCurCtlFlowChar  = (state.vxContext.svxCurCtlFlowChar=="")
+              ? ch : state.vxContext.svxCurCtlFlowChar;
+            stream.next();
+            stream.match(/[a-zA-Z_0-9]+/);
+            break;
+          }
+          if (state.vxContext.svxCurCtlFlowChar != vxCurCtlFlowCharValueAtStart) { // flow change
+            vxIndent=thisMode.vxHooks["svxGenIndent"](stream, state, thisMode);
+            state.vxContext.vxIndentRq = vxIndent;
+          }
+        }
+        return style;
+      },
+      "startStateContext": function(thisMode) {
+        this.svxCurCtlFlowChar = "";
+        this.svxPrevCtlFlowChar = "";
+        this.svxPrevPrevCtlFlowChar = "";
+        this.vxCodeActive = thisMode.vxModeIsOn;
+        this.vxIndentRq = 0;
+      },
+      "token": function(stream, state, _thisMode) {
+        if (state.vxContext.vxCodeActive==true && stream.sol() && state.vxContext.svxCurCtlFlowChar != "") {
+          state.vxContext.svxPrevPrevCtlFlowChar = state.vxContext.svxPrevCtlFlowChar;
+          state.vxContext.svxPrevCtlFlowChar = state.vxContext.svxCurCtlFlowChar;
+          state.vxContext.svxCurCtlFlowChar  = "";
+        }
+      },
+      "svxGenIndent": function(stream, state, thisMode) {
```

Comment:
How about making this a function, rather than putting it in the hooks object, though no dynamic dispatch is needed?